stellargraph/core/graph_networkx.py

```diff
@@ -709,16 +709,29 @@ def number_of_edges(self) -> int:
     def nodes(self) -> Iterable[Any]:
         return self._graph.nodes()
 
-    def edges(self, include_edge_type=False) -> Iterable[Any]:
+    def edges(
+        self, include_edge_type=False, include_edge_weight=False
+    ) -> Iterable[Any]:
         if include_edge_type:
             # returns triples of format (node 1, node 2, edge type)
-            return (
+            edges = (
                 (src, dst, self._get_edge_type(data))
                 for src, dst, data in self._graph.edges(data=True)
             )
         else:
             # returns pairs of format (node 1, node 2)
-            return self._graph.edges()
+            edges = self._graph.edges()
+
+        if include_edge_weight:
+            return (
+                edges,
+                [
```

Comment:
How about returning this as a numpy array, so that we're doing #718 for all new APIs (and matching the new implementation)? Also, splitting out the computation into a separate variable may make this code slightly less explosively formatted:

```python
weights = np.array([data.get(self._edge_weight_label) for _, _, data in self._graph.edges(data=True)])
return edges, weights
```